### PR TITLE
Migrate edit-widgets store to thunks

### PR DESCRIPTION
**Pull request by adamziel**
This PR migrates the edit-widgets data store to thunks, namely actions and resolvers. It also dependency on `@wordpress/data-controls` and on any other controls for that matter.

**How to test:**
Open the widgets editor and confirm it loads and saves widgets as expected.

### CODE REVIEW

File: packages/edit-widgets/src/store/resolvers.js [hunk 1]
```diff
@@ -18,16 +18,11 @@ import {
...
-	yield resolveWidgetAreas( query );
-	const widgetAreas = yield select(
-		'core',
-		'getEntityRecords',
-		KIND,
-		WIDGET_AREA_ENTITY_TYPE,
-		query
-	);
+	const widgetAreas = await registry
+		.resolveSelect( 'core' )
```
Comment (Mamaduka) [inline packages/edit-widgets/src/store/resolvers.js]:
```suggestion
		.resolveSelect( coreStore )
```

It looks like the ESLint rule doesn't work in this scenario.

**Review by Mamaduka — APPROVED** [review]
The widget screen works as expected in my local tests.

File: packages/edit-widgets/src/store/controls.js [hunk 2]
```diff
@@ -1,181 +0,0 @@
...
-}
-
-/**
- * Resolves navigation post for given menuId.
- *
- * @see selectors.js
- * @param {number} menuId Menu ID.
- * @return {Object} Action.
- */
-export function getNavigationPostForMenu( menuId ) {
```
Comment (noisysocks) [inline packages/edit-widgets/src/store/controls.js:64]: How'd this get in here! 😛

**Review by noisysocks — APPROVED** [review]
Code looks good 👍 Glad that we're throwing `controls.js` in the bin. I didn't test this locally but if E2E tests pass then I'm happy since Kai did a pretty good job covering all the main widget flows.

I noticed we're missing doc comments for most of the `export`ed actions that are being modified here. If you want bonus brownie points, maybe add these missing comments?

**Comment by adamziel** [comment]
> If you want bonus brownie points, maybe add these missing comments?
I do want brownie points :-) Let's tackle that in a follow-up PR

**Pull request merged**